theano/tensor/blas.py

```diff
@@ -1918,19 +1918,19 @@ def local_dot22_to_ger_or_gemv(node):
             # x and y are both vectors so this qualifies for a sdot / ddot
             # TODO: Theano doesn't have a sdot, but gemv is better than _dot22
             xv = x.dimshuffle(1)
-            zeros = T.AllocEmpty(x.dtype)(1)
+            zeros = T.zeros([1], x.dtype)
```

Comment:
This will slow down some computation. The problem happen only for some blas implementation. Others won't do the mul by 0, so they don't have the problem.

We have a work around that will detect if the blas used need this 0s of the the input or not. This allow to use the slow version only if needed. So a better fix would be to reuse that. It isn't very hard, can you do it?

in the file tensor/blas_c.py, there is a function check_force_gemv_init(). Can you move it to this file and use it here to choose to zeros or not?

Also, can you test it on your side? It is hard for us to reproduice this and test it locally.

thanks